recipes/opencolorio/all/conanfile.py

```diff
@@ -49,96 +40,127 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("expat/2.4.8")
-        self.requires("openexr/2.5.7")
+        self.requires("expat/2.5.0")
+        if Version(self.version) < "2.2.0":
+            self.requires("openexr/2.5.7")
+        else:
+            self.requires("openexr/3.1.7")
         self.requires("yaml-cpp/0.7.0")
-        if tools.Version(self.version) < "2.0.0":
+        if Version(self.version) < "2.0.0":
             self.requires("tinyxml/2.6.2")
         else:
-            self.requires("pystring/1.1.3")
+            self.requires("pystring/1.1.4")
+        self.requires("imath/3.1.6")
+        if Version(self.version) >= "2.2.0":
+            self.requires("minizip-ng/3.0.7")
+
         # for tools only
-        self.requires("lcms/2.13.1")
+        self.requires("lcms/2.14")
         # TODO: add GLUT (needed for ociodisplay tool)
 
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 11)
+            check_min_cppstd(self, 11)
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        # opencolorio>=2.2.0 requires minizip-ng with with_zlib
+        if Version(self.version) >= "2.2.0" and \
+            not self.dependencies["minizip-ng"].options.get_safe("with_zlib", False):
+            raise ConanInvalidConfiguration(f"{self.ref} requires minizip-ng with with_zlib = True.")
```

Comment:
Since with_zlib option is removed in minizp-ng recipe if apple os, it means that opencolorio >= 2.2.0 always raises for apple os (indirectly) without any way for consumers to change that. Are you sure that option requirement of minizip-ng is that strict?

Reply:
I can confirm that the requirement is correct, opencolorio does not build if minizip-ng doesn't provide the header `mz_strm_zlib.h` (https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/dddbee023dc2d4c21b16fe4f79ca1374c19ecbc0/src/OpenColorIO/OCIOZArchive.cpp#L27) which is only present if minizip-ng is compiled with zlib. #20521 provides a better error message.